File: app/src/main/java/com/willkernel/kotlinapp/test/BaseFragment.java
package com.willkernel.kotlinapp.test;

import android.os.Bundle;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import androidx.annotation.Nullable;
import com.trello.rxlifecycle2.components.RxFragment;

/**
 * Created by willkernel
 * on 2019/3/31.
 */
public abstract class BaseFragment extends RxFragment {

    private View rootView;

    @Nullable
    @Override
    public View onCreateView(LayoutInflater inflater, @Nullable ViewGroup container, Bundle savedInstanceState) {
        super.onCreateView(inflater, container, savedInstanceState);
        if (rootView == null) {
            rootView = inflater.inflate(this.getLayoutId(), container, false);
            init(savedInstanceState);
        }
        ViewGroup viewGroup= (ViewGroup) rootView.getParent();
        if(viewGroup!=null){
            viewGroup.removeView(rootView);
        }
        return rootView;
    }

    protected abstract int getLayoutId();

    protected abstract void init(Bundle savedInstanceState);
}

File: app/src/main/java/com/willkernel/kotlinapp/base/BaseFragment.kt
package com.willkernel.kotlinapp.base

import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.annotation.LayoutRes
import androidx.fragment.app.Fragment
import com.willkernel.kotlinapp.view.MultipleStatusView

/**
 * Created by willkernel
 * on 2019/4/2.
 */
abstract class BaseFragment : Fragment() {
    protected val TAG: String = javaClass.name
    /**
     * 多种状态的 View 的切换
     */
    protected var mLayoutStatusView: MultipleStatusView? = null
    override fun onCreateView(inflater: LayoutInflater, container: ViewGroup?, savedInstanceState: Bundle?): View? {
        super.onCreateView(inflater, container, savedInstanceState)
        return inflater.inflate(getLayoutId(), null)
    }

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)
        initView()
        lazyLoad()
    }

    abstract fun initView()

    abstract fun lazyLoad()

    @LayoutRes
    abstract fun getLayoutId(): Int
}